src/js/player.js

```diff
@@ -1124,29 +1127,28 @@ class Player extends Component {
    *
    * @fires Player#firstplay
    *
-   * @param {boolean} hasStarted
+   * @param {boolean} request
    *        - true: adds the class
    *        - false: remove the class
    *
    * @return {boolean}
-   *         the boolean value of hasStarted
-   */
-  hasStarted(hasStarted) {
-    if (hasStarted !== undefined) {
-      // only update if this is a new value
-      if (this.hasStarted_ !== hasStarted) {
-        this.hasStarted_ = hasStarted;
-        if (hasStarted) {
-          this.addClass('vjs-has-started');
-          // trigger the firstplay event if this newly has played
-          this.trigger('firstplay');
-        } else {
-          this.removeClass('vjs-has-started');
-        }
-      }
-      return;
+   *         the boolean value of hasStarted_
+   */
+  hasStarted(request) {
+    if (request === undefined || request === this.hasStarted_) {
```

Comment:
A comment explaining that this is used as a getter or if the value isn't changing, we should just return what we have would be good